monitoring.go

```diff
@@ -225,7 +225,9 @@ func (sc *snowflakeConn) waitForCompletedQueryResultResp(
 	resultPath string,
 ) (*execResponse, error) {
 	// if we already have the response; return that
-	if cachedResponse, ok := sc.execRespCache.load(resultPath); ok {
+	cachedResponse, ok := sc.execRespCache.load(resultPath)
+	logger.WithContext(ctx).Errorf("use cache: %v", ok)
+	if ok {
```

Comment:
We should do/log more here. 

If we don't use the cache, we are back to the default driver behavior. But if we use it, then we need to validate why it doesn't work.